distsql/select_result_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package distsql
+
+import (
+	"fmt"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/tidb/sessionctx/stmtctx"
+	"github.com/pingcap/tidb/util/execdetails"
+	"github.com/pingcap/tidb/util/mock"
+	"github.com/pingcap/tipb/go-tipb"
+)
+
+func (s *testSuite) TestUpdateCopRuntimeStats(c *C) {
+	ctx := mock.NewContext()
+	ctx.GetSessionVars().StmtCtx = new(stmtctx.StatementContext)
+	sr := selectResult{ctx: ctx}
+	c.Assert(ctx.GetSessionVars().StmtCtx.RuntimeStatsColl, IsNil)
+	sr.updateCopRuntimeStats("a")
+
+	ctx.GetSessionVars().StmtCtx.RuntimeStatsColl = execdetails.NewRuntimeStatsColl()
+	t := uint64(1)
+	sr.selectResp = &tipb.SelectResponse{
+		ExecutionSummaries: []*tipb.ExecutorExecutionSummary{
+			{TimeProcessedNs: &t, NumProducedRows: &t, NumIterations: &t},
+		},
+	}
+	c.Assert(len(sr.selectResp.GetExecutionSummaries()) != len(sr.copPlanIDs), IsTrue)
+	sr.updateCopRuntimeStats("callee")
+
+	sr.copPlanIDs = []fmt.Stringer{copPlan{}}
+	c.Assert(ctx.GetSessionVars().StmtCtx.RuntimeStatsColl, NotNil)
+	c.Assert(len(sr.selectResp.GetExecutionSummaries()), Equals, len(sr.copPlanIDs))
+	sr.updateCopRuntimeStats("callee")
+	for _, execSum := range sr.selectResp.GetExecutionSummaries() {
+		c.Assert(*execSum.NumIterations, Equals, uint64(1))
+		c.Assert(*execSum.NumProducedRows, Equals, uint64(1))
+		c.Assert(*execSum.TimeProcessedNs, Equals, uint64(1))
```

Comment:
Should we check the values in `ctx.GetSessionVars().StmtCtx.RuntimeStatsColl` instead of `sr.selectResp.GetExecutionSummaries()`?